src/main/java/luffy/LuffyParser.java

```diff
@@ -0,0 +1,113 @@
+                break;
+            }
+
+            if (taskAndDeadline[1].startsWith("by") || taskAndDeadline[1].startsWith("By")) {
+                String dateAndTime = taskAndDeadline[1].substring(3).trim();
+
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+                LocalDateTime timeObj = LocalDateTime.parse(dateAndTime, formatter);
+                String dateTime = timeObj.format(DateTimeFormatter.ofPattern("MMM dd yyyy HHmm"));
+                deadlineTask = new Deadline(taskAndDeadline[0].trim(), dateTime + "HRS", false);
+                return new AddCommand(deadlineTask);
+            } else {
+                deadlineTask = new Deadline(taskAndDeadline[0].trim(), taskAndDeadline[1].trim(), false);
+                return new AddCommand(deadlineTask);
+            }
+
+        case "event":
+            if (commandDetails.length == 1) {
+                break;
+            }
+            String[] eventDetails = fullCommand.substring(6).split("/");
+            try {
+                checkValidArguments(eventDetails, 3);
+            } catch (LuffyException e) {
+                System.out.println(e.getMessage());
+                break;
+            }
+            Event eventTask = new Event(eventDetails[0].trim(), eventDetails[1].trim(), eventDetails[2], false);
+            return new AddCommand(eventTask);
+
+        case "find":
+            return new FindCommand(commandDetails[1]);
+
+        case "mark":
+            int markIndex = Integer.parseInt(fullCommand.substring(5)) - 1;
+            return new MarkCommand(markIndex);
+
+        case "unmark":
+            int unmarkIndex = Integer.parseInt(fullCommand.substring(7)) - 1;
+            return new UnmarkCommand(unmarkIndex);
+
+        case "delete":
+            int deleteIndex = Integer.parseInt(fullCommand.substring(7)) - 1;
+            return new DeleteCommand(deleteIndex);
+
+        case "list":
+            return new ListCommand();
+
+        case "bye":
+            return new ExitCommand();
+
+        default:
+            return new InvalidCommand();
+        }
+        return new InvalidCommand();
+    }
+
+    /**
+     * This method checks if the length of the text list matches the expected length
+     *
+     * @param textList the split command for checking
+     * @param expectedLength the expected length of the textList parameter
+     * @throws LuffyException If the expectedLength does not equal to the length of the textList
+     */
+    public static void checkValidArguments(String[] textList, int expectedLength) throws LuffyException {
+        if (textList.length != expectedLength && expectedLength == 2) {
+            throw new LuffyException("Your deadline task requires a deadline!");
+        } else if (textList.length != expectedLength && expectedLength == 3) {
+            throw new LuffyException("Your event task requires a start time and end time!");
+        }
```

Comment:
Good catch! Its always good to ensure there is exactly 2 or 3 commands because more or less than required commands could be given for either command.